packages/@lwc/engine-server/src/utils/validate-style-text-contents.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { parseFragment, DocumentFragment } from 'parse5';
+
+function isSingleStyleNodeContainingSingleTextNode(node: DocumentFragment) {
+    if (node.childNodes.length !== 1) {
+        return false;
+    }
+
+    const style = node.childNodes[0];
+
+    if (style.nodeName !== 'style' || style.childNodes.length !== 1) {
+        return false;
+    }
+
+    const textNode = style.childNodes[0];
+
+    return textNode.nodeName === '#text';
+}
```

Comment:
There is probably a super-clever way to detect that the contents of a `<style>` tag are going to result in an XSS vulnerability, but the most bulletproof solution I could think of is to parse the whole thing in parse5 and see if it results in the expected single `<style>` tag.

There is probably also a way to optimize this (e.g. check that the string actually contains any special characters), but again, the most bulletproof thing is to just check everything. If this turns out to be a perf problem later, then we can optimize it then.

Reply:
parse5 doesn't have built-in sanitization right? We're relying on
```
(single <style> tag + single text node) === SAFE
```
?

Reply:
Exactly, yes. This is why I have a test where the CSS file contains `</style>`. I can add a few more to catch a few more scenarios as well.

Reply:
Done: d5242e75d